### PR TITLE
Retry on 408, 500, 502, 504 status codes

### PR DESCRIPTION
**Pull request by pakrym**
Align with other azure core libraries.

### CODE REVIEW

**Review by mikeharder — APPROVED** [review]

File: sdk/core/Azure.Core/src/ResponseClassifier.cs [hunk 1]
```diff
@@ -16,7 +16,14 @@ public class ResponseClassifier
...
         {
-            return message.Response.Status == 429 || message.Response.Status == 503;
+            var responseStatus = message.Response.Status;
+
+            return responseStatus == 408 ||
+                   responseStatus == 429 ||
+                   responseStatus == 500 ||
+                   responseStatus == 502 ||
+                   responseStatus == 503 ||
+                   responseStatus == 504;
```
Comment (tg-msft) [inline sdk/core/Azure.Core/src/ResponseClassifier.cs]: Small request for anyone who ends up here when they're having a very bad day...  could we add a comment with the friendly name of each of these status codes right next to it?

**Comment by heaths** [comment]
Resolves Azure/azure-sdk#975

**Review by tg-msft — APPROVED** [review]
Looks good to me.  We should run it by Krzysztof to weigh in on whether there's any chance this could be considered a behavioral breaking change.

File: sdk/core/Azure.Core/src/ResponseClassifier.cs [hunk 2]
```diff
@@ -16,7 +16,14 @@ public class ResponseClassifier
         /// </summary>
         public virtual bool IsRetriableResponse(HttpMessage message)
         {
-            return message.Response.Status == 429 || message.Response.Status == 503;
+            var responseStatus = message.Response.Status;
+
+            return responseStatus == 408 ||
```
Comment (heaths) [inline sdk/core/Azure.Core/src/ResponseClassifier.cs]: Would a switch be faster here? Or even comparison logic like other languages, though I doubt that (seems more convenience in code done that way).

**Review by heaths — APPROVED** [review]

**Review by mikeharder — CHANGES_REQUESTED** [review]

**Review by mikeharder — APPROVED** [review]

**Comment by pakrym** [comment]
> Looks good to me. We should run it by Krzysztof to weigh in on whether there's any chance this could be considered a behavioral breaking change.

He's fine with the change.

**Pull request merged**